src/Microsoft.AspNetCore.Razor.Language/DefaultRazorCSharpLoweringPhase.cs

```diff
@@ -60,6 +60,7 @@ protected override void ExecuteCore(RazorCodeDocument codeDocument)
 
             var diagnostics = new List<RazorDiagnostic>();
             diagnostics.AddRange(syntaxTree.Diagnostics);
+            diagnostics.AddRange(irDocument.GetAllDiagnostics());
```

Comment:
> I feel that the only errors that make sense on the CSharpDocument itself are those that were created during lowering.

This is wrong. Each one of these is a step in a process. As the process continues we get more information